File: ios/WebViewFramework/WebViewFramework/GogolookWebViewContract.swift
//
//  GogolookWebContainer.swift
//  WebViewFramework
//
//  Created by Willy on 2025/5/26.
//

protocol GogolookWebViewContract: AnyObject {
    func initWebContainer(
        webSecurityConfig: WebSecurityConfig,
        webContainerConfig: WebContainerConfig,
        webBridgeCallback: WebBridgeCallback
    )
    
    func loadUrl(_ url: String)
    func syncConfigToWeb(_ config: WebContainerConfig)
    func cleanUp()
}

File: android/WebViewFramework/app/src/main/java/com/gogolook/webviewframework/GogolookWebViewContract.kt
package com.gogolook.webviewframework

interface GogolookWebViewContract {
    fun initWebContainer(
        webSecurityConfig: WebSecurityConfig,
        webContainerConfig: WebContainerConfig,
        webBridgeCallback: WebBridgeCallback,
    )

    fun loadUrl(url: String)
    fun syncConfigToWeb(webContainerConfig: WebContainerConfig)
    fun cleanUp()
}